vita3k/modules/SceGxm/SceGxm.cpp

```diff
@@ -4191,6 +4191,9 @@ EXPORT(void, sceGxmSetViewportEnable, SceGxmContext *context, SceGxmViewportMode
     // Set viewport to enable/disable, no additional offset and scale to set.
     if (context->state.viewport.enable != enable) {
         context->state.viewport.enable = enable;
+        if (!context->renderer->current_render_target) {
```

Comment:
This should be checked [here](https://github.com/Vita3K/Vita3K/blob/f43ac8765ea1a28bb0270092dadab9c900f991de/vita3k/renderer/src/state_set.cpp#L150) instead.

Moreover, this does not check that this happens within a scene (current_render_target never gets set back to null after being set). This will cause the first scene in each game to completly ignore the viewport (and even the scenes after could depending on race conditions).

Reply:
I modified it to check context->state.active instead of current_render_target to verify if we are within a Scene when executing the command, as it seems more appropriate.
Apart from that, as I understand it, sceGxmBeginScene calls sceGxmSetDefaultRegionClipAndViewport during initialization, and sceGxmSetDefaultRegionClipAndViewport calls set_viewport_real and set_viewport_flat depending on whether the viewport is enabled, so it shouldn't be problematic.
This is because when a Scene is initialized, the Viewport is reset based on its Enable status.